dkg-gadget/src/worker.rs

```diff
@@ -923,15 +932,39 @@ where
 			// If the next rounds have stalled, we restart similarly to above.
 			if let Some(rounds) = self.next_rounds.read().as_ref() {
 				debug!(target: "dkg_gadget::worker", "🕸️  Status: {:?}, Now: {:?}, Started At: {:?}, Timeout length: {:?}", rounds.status, header.number(), rounds.started_at, KEYGEN_TIMEOUT);
-				if rounds.keygen_has_stalled(*header.number()) {
-					debug!(target: "dkg_gadget::worker", "🕸️  QUEUED DKG STALLED: round {:?}", queued.id);
+				let keygen_stalled = rounds.keygen_has_stalled(*header.number());
+				let (current_attmp, max, should_retry) = {
+					// check how many authorities are in the next best authorities
+					// and then check the signature threshold `t`, if `t+1` is greater than the
+					// number of authorities and we still have not reached the maximum number of
+					// retries, we should retry the keygen
+					let n = next_best.len();
+					let t = self.get_signature_threshold(header) as usize;
```

Comment:
If this is next keygen then you need to use next signature threshold.

Reply:
it should be the current `t` since we will create a misbehaviour reports on the current session/round. and Since we do not want the Misbehaviour reports to be useless, which would be the case if the number of reports (which should be equal to `t+1`) is less than `t+1`. But I will modify that, it should only be if `n` equal to `t+1` that means if we did try to misbehaviour report someone it will fail. Example: n = 2 and t = 1, we need at least `t+1 = 2` reports to report one of these nodes, and since it will only be one report.

Reply:
my bad, it should be the next signature threshold, I found that in the DKG pallet here: https://github.com/webb-tools/dkg-substrate/blob/826d389aa57f67a3f87ef790b021e714e309a815/pallets/dkg-metadata/src/lib.rs#L911-L916